### ISSUE TITLE
infusion events need to be handled correctly in DataFrame 

### ISSUE THREAD

**Issue by vjd** [PumasAI/Pumas.jl#633]
Consider this simple example -

```julia
julia> sd = DosageRegimen(2000, time=0, rate=-2)
julia> sd
DosageRegimen(1×8 DataFrame
│ Row │ time    │ cmt   │ amt     │ evid │ ii      │ addl  │ rate    │ ss   │
│     │ Float64 │ Int64 │ Float64 │ Int8 │ Float64 │ Int64 │ Float64 │ Int8 │
├─────┼─────────┼───────┼─────────┼──────┼─────────┼───────┼─────────┼──────┤
│ 1   │ 0.0     │ 1     │ 2000.0  │ 1    │ 0.0     │ 0     │ -2.0    │ 0    │)
```
Now put this into a framework of a `Subject` and check the events of the subject as below

```julia
s1 = Subject(id=1,evs=sd, cvs=(Wt=70,))
julia> s1.events
2-element Array{Pumas.Event,1}:
 End of infusion event
  dose amount = 2000.0
  dose time = -1000.0
  compartment = 1
  duration specified by model
  interdose interval = 0.0
  infusion start time = 0.0
  end of infusion

 Dose event
  dose amount = 2000.0
  dose time = 0.0
  compartment = 1
  duration specified by model
  interdose interval = 0.0
  infusion start time = 0.0
```
Notice how that for infusion there is an initial off event followed by an on event resulting in an array of 2 events. Obviously the `DataFrame` method on this results in

```julia
julia> DataFrame(s1.events)
2×10 DataFrame
│ Row │ amt     │ time    │ evid │ cmt   │ rate    │ duration │ ss   │ ii      │ base_time │ rate_dir │
│     │ Float64 │ Float64 │ Int8 │ Int64 │ Float64 │ Float64  │ Int8 │ Float64 │ Float64   │ Int8     │
├─────┼─────────┼─────────┼──────┼───────┼─────────┼──────────┼──────┼─────────┼───────────┼──────────┤
│ 1   │ 2000.0  │ -1000.0 │ -1   │ 1     │ -2.0    │ -1000.0  │ 0    │ 0.0     │ 0.0       │ -1       │
│ 2   │ 2000.0  │ 0.0     │ 1    │ 1     │ -2.0    │ -1000.0  │ 0    │ 0.0     │ 0.0       │ 1        │
```

This is what is being used in the `SimulatedObservation` https://github.com/PumasAI/Pumas.jl/blob/master/src/models/simulated_observations.jl#L16-L82

So, if I simulate from a model using this subject, I get:

```julia
obs = simobs(iv, s1, param, obstimes=sd_obstimes)
```

```julia
julia> DataFrame(obs)
┌ Error: Error adding value to column :dv.
└ @ DataFrames ~/.julia/packages/DataFrames/Iyo5L/src/dataframe/dataframe.jl:1295
15×7 DataFrame
│ Row │ time    │ dv       │ amt     │ evid │ cmt     │ rate    │ Wt    │
│     │ Float64 │ Float64⍰ │ Float64 │ Int8 │ Int64⍰  │ Float64 │ Int64 │
├─────┼─────────┼──────────┼─────────┼──────┼─────────┼─────────┼───────┤
│ 1   │ -1000.0 │ missing  │ 2000.0  │ -1   │ 1       │ -2.0    │ 70    │
│ 2   │ 0.0     │ 0.0      │ 2000.0  │ 1    │ 1       │ -2.0    │ 70    │
│ 3   │ 0.0     │ 0.0      │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
│ 4   │ 0.25    │ 12.104   │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
│ 5   │ 0.5     │ 23.2842  │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
│ 6   │ 0.75    │ 35.3718  │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
│ 7   │ 1.0     │ 46.4375  │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
│ 8   │ 2.0     │ 34.1136  │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
│ 9   │ 4.0     │ 21.6819  │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
│ 10  │ 8.0     │ 7.64589  │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
│ 11  │ 12.0    │ 2.89452  │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
│ 12  │ 16.0    │ 1.0638   │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
│ 13  │ 20.0    │ 0.364102 │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
│ 14  │ 22.0    │ 0.234484 │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
│ 15  │ 24.0    │ 0.141476 │ 0.0     │ 0    │ missing │ 0.0     │ 70    │
```

Since off-events can happen anytime based on the design, there should be an easier way to tag such events to not get propagated into post-processing frameworks such as the DataFrame here. 

Or did I just think too much of this problem 🤷‍♂

**Comment by ChrisRackauckas** [comment]
There's already an int for whether it's an off event, so the Subject/Population/SimulatedObservation etc. dataframe constructors can just check for that and drop the event if that's `-1`. @pkofod

**Comment by pkofod** [comment]
I understand how to handle it, but I’m not sure what this means. What does rate=-2 mean?

**Comment by ChrisRackauckas** [comment]
`rate=-2` means that the duration is specified by a dosing control parameter.

**Comment by pkofod** [comment]
This was fixed in #630
